symfony/client/src/orangehrmAdminPlugin/pages/workShift/EditWorkShift.vue

```diff
@@ -0,0 +1,218 @@
+  validTimeFormat,
+} from '@orangehrm/core/util/validation/rules';
+import {diffInTime, secondsTohhmm} from '@/core/util/helper/datefns';
+import WorkShiftEmployeeAutocomplete from '@/orangehrmAdminPlugin/components/WorkShiftEmployeeAutocomplete';
+
+const workShiftModel = {
+  id: '',
+  name: '',
+  hoursPerDay: '',
+  startTime: null,
+  endTime: null,
+  empNumbers: [],
+};
+const arrEmp = [];
+
+export default {
+  props: {
+    workShiftId: {
+      type: Number,
+      required: true,
+    },
+  },
+  components: {
+    'work-shift-employee-autocomplete': WorkShiftEmployeeAutocomplete,
+  },
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      '/api/v2/admin/work-shifts',
+    );
+    return {
+      http,
+    };
+  },
+  data() {
+    return {
+      isLoading: false,
+      workShift: {...workShiftModel},
+      rules: {
+        name: [required, shouldNotExceedCharLength(50)],
+        endTime: [
+          validTimeFormat,
+          endTimeShouldBeAfterStartTime(
+            () => this.workShift.startTime,
+            'To time should be after from time',
+          ),
+        ],
+      },
+    };
+  },
+  methods: {
+    onSave() {
+      this.isLoading = true;
+      this.workShift.empNumbers.forEach(function(employee) {
+        arrEmp.push(employee.id);
+      });
+
+      const payload = {
+        name: this.workShift.name,
+        hoursPerDay: parseFloat(
+          diffInTime(this.workShift.startTime, this.workShift.endTime) / 3600,
+        ).toFixed(2),
+        startTime: this.workShift.startTime,
+        endTime: this.workShift.endTime,
+        empNumbers: arrEmp,
+      };
+      this.http
+        .update(this.workShiftId, payload)
+        .then(() => {
+          return this.$toast.updateSuccess();
+        })
+        .then(() => {
+          this.onCancel();
+        });
+    },
+    onCancel() {
+      navigate('/admin/viewWorkShifts');
+    },
+  },
+  beforeMount() {
+    this.isLoading = true;
+    this.http
+      .get(this.workShiftId)
+      .then(response => {
+        const {data} = response.data;
+        this.workShift.id = data.id;
+        this.workShift.name = data.name;
+        this.workShift.hoursPerDay = data.hours_per_day;
+        this.workShift.startTime = data.start_time;
+        this.workShift.endTime = data.end_time;
+        this.workShift.empNumbers = data.employee;
+        return this.http.getAll();
+      })
+      .then(response => {
+        const {data} = response.data;
+        this.rules.name.push(v => {
+          const index = data.findIndex(item => item.name == v);
+          if (index > -1) {
+            const {id} = data[index];
+            return id != this.workShift.id ? 'Already exists' : true;
+          } else {
+            return true;
+          }
+        });
+      })
+      .finally(() => {
+        this.isLoading = false;
+      });
+  },
+  computed: {
+    selectedTimeDuration() {
+      return secondsTohhmm(
```

Comment:
Please make sure the spec requires `hh:mm` format if you're using this method.

Reply:
done